### PR TITLE
fix(sdk): update init to support promises

### PR DESCRIPTION
**Pull request by crimeminister**
# Description

Initialization multimethods (as used by `integrant`) that return promises end up with the promise merged into the system map, rather than the resolved data being used. We need to update our init to support promises at a minimum, and futures would be nice as well if it doesn't add much complexity.

Some options to consider include:
- `donut-power/system`; this feature on the roadmap but not currently supported
- `athos/aintegrant`; potential drop-in replacement?
- `juxt/clip`; apparently supports this now, check how much rework entailed by switching
- send change upstream to `integrant` (check if author is interested in this contribution)
- write some straight-line initialization code and forego DI and other features arising out of an init library

This also lays some groundwork for eventual adoption of a cross-platform promise library with some additional useful features. In particular, some of the wallet code, and the `ddt` command handlers now deal in promises as a step towards making things more work more consistently across the code base.

## Type of change

- [x] Bug-fix
- [x] Clean-up and refactoring

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation website
- [x] I have made corresponding changes to the literal docs
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tangrammer** [comment]
Initialising ig system in a different async context  (`go`) 
https://gist.github.com/tangrammer/f200c545525f0c527b03abb88f1bee43#file-integrant-cljc

**Review by Cosmin-Parvulescu — APPROVED** [review]

**Comment by tangrammer** [comment]
Hi @crimeminister 
 before adding last detail of "new config key/val to mock ipfs-client if there is no one around" ... could we merge first https://github.com/kubelt/kubelt/pull/334 ? I wouldn't like to have conflict problems again on 334 😬

**Comment by tangrammer** [comment]
BTW: Added a couple of changes to fix and let exceptions flowing up in http node/ipfs calls
- [src/main/com/kubelt/ipfs/client.cljc](https://github.com/kubelt/kubelt/pull/340/files#diff-214d6d8286cfc003449acb8ec77dc24ef2b4e500d3776708caecace82f898a62)
- [src/main/com/kubelt/lib/http/node.cljs](https://github.com/kubelt/kubelt/pull/340/files#diff-459b20738d3904450efd4e596ab2902b614d9582e9873d72edd6661d15c1f6a1)

**Comment by tangrammer** [comment]
@crimeminister 
Rebased this branch with main one, so right now it's aligned with last refactor config changes

Regarding https://github.com/kubelt/kubelt/pull/340#issuecomment-1110789457 I'll create a different branch for achieving that

**Comment by tangrammer** [comment]
@maurerbot i think so :)

**Comment by drew-patel** [comment]
@crimeminister I was speaking to @maurerbot about this over the weekend.

We don't run integration tests for `dapp` on `sdk` or `ddt` changes during the PR stage. We do however run them on merge. That is was caused this: https://github.com/kubelt/kubelt/issues/364

Is it cool to update the github workflow file to run `dapp` tests for any change? They technically fall into the integration test category now.

**Comment by crimeminister** [comment]
> @crimeminister I was speaking to @maurerbot about this over the weekend.
> 
> We don't run integration tests for `dapp` on `sdk` or `ddt` changes during the PR stage. We do however run them on merge. That is was caused this: #364
> 
> Is it cool to update the github workflow file to run `dapp` tests for any change? They technically fall into the integration test category now.

Please do run the `dapp` tests, the longer the wall of green, the happier I get.

File: src/main/com/kubelt/lib/integrant.cljc [hunk 1]
```diff
@@ -0,0 +1,44 @@
+(ns com.kubelt.lib.integrant
+  "extending ig.core/init and ig.core/build in order to support/realise js/promises
```
Comment (drew-patel) [inline src/main/com/kubelt/lib/integrant.cljc:2]: This is dope!

Comment (crimeminister) [reply src/main/com/kubelt/lib/integrant.cljc:2]: @tangrammer did some magic there, agreed! Would be great if there was interest in landing this upstream.

**Review by alfl — APPROVED** [review]
👍

**Pull request merged**